oar/core/advisory.py

```diff
@@ -629,3 +661,100 @@ def has_blocking_secruity_alert(self):
             return blocking
         else:
             return False
+        
+    def get_unhealthy_builds(self):
+        """
+        Get unhealthy builds of advisory.
+
+        Returns:
+            list: Unhealty builds {}.
+        """
+        nvrs = self.get_build_nvrs()
+
+        unhealthy_builds = []
+
+        for n in nvrs:
+            build_grades = self.get_build_grades(n)
+            for bg in build_grades:
+                if not util.isGradeHealthy(bg["grade"]):
+                    unhealthy_builds.append(bg)
+
+        return unhealthy_builds
+        
+    def get_build_grades(self, nvr):
+        """
+        Get all architecture grades of build.
+
+        Returns:
+            list: All architecture grades of build {nvr, architecture, grade}.
+        """
+        nvr_url = "https://pyxis.engineering.redhat.com/v1/images/nvr/%s?include=data.freshness_grades&include=data.architecture" % nvr
+        resp = requests.get(nvr_url, auth=self._auth, verify=self.ssl_verify)
+
+        nvr_grades = []
+
+        if resp.ok:
+            # TODO should we skip empty data? 
+
+            for arch in resp.json()["data"]:
+                grade = None
+                for fg in arch["freshness_grades"]:                    
+                    new_grade = fg["grade"]
+                    start_date = parser.parse(fg["start_date"]).replace(tzinfo=None)
```

Comment:
we'd better to get parsed and now with same tz. the datetime format from pyxis api is utc
```
from datetime import datetime
from datetime import timezone
from dateutil import parser

dt_str = "2024-10-24T21:26:48.913000+00:00"
start_date = parser.parse(dt_str)
print(start_date < datetime.now(timezone.utc))
```

Reply:
Fixed. Please check.